### DISCUSSION TITLE
Should libraries using Undici ship Undici as peerDependency or dependency

### DISCUSSION THREAD

**Discussion by semoal** [nodejs/undici#3936] [Q&A] [answered]
I'm migrating a Nodejs library to Undici from `http.Agent`, we'll drop support to Node < 20  and we're thinking if we should put Undici as dependency or peer-dependency, also I've installed Undici ^7.x... is it too new? or should we stay in 6.x until 24 is LTS?

Sorry if this question is too dumb, but I'm not 100% sure.

**Comment by metcoder95** [comment, marked as answer]
If your library is a direct dependent of `undici`, the best is to set it as direct `dependency`. Specially if you are attempting to make usage of `v7` which installers can/cannot be willing to set it to that version if they are already using `undici`.

`v7` should be totally fine as is marked as the new current version for the library.

**Reply by semoal** [reply]
Thanks Carlos, really appreciated for the help! ❤️
